Traceback (most recent call last):
  File "/Users/juliafiedler/anaconda3/envs/SLI39/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/juliafiedler/anaconda3/envs/SLI39/lib/python3.9/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/juliafiedler/anaconda3/envs/SLI39/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 162, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/juliafiedler/anaconda3/envs/SLI39/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/juliafiedler/anaconda3/envs/SLI39/lib/python3.9/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/juliafiedler/anaconda3/envs/SLI39/lib/python3.9/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/juliafiedler/anaconda3/envs/SLI39/lib/python3.9/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# plot the monthly maxima
monthly_max = get_extremes(sea_level_data_detrended, method='BM', 
                           extremes_type='high', block_size='30D', min_last_block=0.9, errors='coerce')
annual_max = get_extremes(sea_level_data_detrended, method='BM', extremes_type='high', block_size='365.245D', min_last_block=0.9, errors='coerce')

model.get_extremes(extremes_type='high', method='BM', block_size='365D',min_last_block=min_last_block_ratio, errors='coerce')

fig, ax = plt.subplots(figsize=(12,4))
ax.plot(monthly_max.index, monthly_max, label='Monthly Maxima',linewidth=0.5)
ax.scatter(annual_max.index, annual_max, label='Annual Maxima')
ax.set_xlabel('Time')
ax.set_ylabel('Sea Level [m, MHHW]')
ax.set_title(f'Extreme Water Levels, detrended: {station_name}')

# add the 2, 10, 50, and 100 year return levels in horizontal lines
for i,period in enumerate(return_periods):
    ax.axhline(summary.loc[period, 'Water Level (m, MHHW)'], label=f'{period} yr RL', color=colors[i],linewidth=0.5)

# add bold black line at 0 for MHHW
ax.axhline(0, color='black', linewidth=2)
ax.legend()
# put the legend at the bottom of the plot in one row
ax.legend(loc='upper center', bbox_to_anchor=(0.5, -0.2), ncol=3)
ax.set_xlim([rsl_hourly['time'].values[0], rsl_hourly['time'].values[-1]])

# save the figure
figname = 'SL_extremes_monthly_annual_maxima_' + station_name + '.png'
plt.savefig(output_dir / figname, dpi=300, bbox_inches='tight')
------------------

----- stderr -----
/Users/juliafiedler/anaconda3/envs/SLI39/lib/python3.9/site-packages/pyextremes/extremes/block_maxima.py:138: NoDataBlockWarning: 25 blocks contained no data
  warnings.warn(
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[24], line 17[0m
[1;32m     15[0m [38;5;66;03m# add the 2, 10, 50, and 100 year return levels in horizontal lines[39;00m
[1;32m     16[0m [38;5;28;01mfor[39;00m i,period [38;5;129;01min[39;00m [38;5;28menumerate[39m(return_periods):
[0;32m---> 17[0m     ax[38;5;241m.[39maxhline(summary[38;5;241m.[39mloc[period, [38;5;124m'[39m[38;5;124mWater Level (m, MHHW)[39m[38;5;124m'[39m], label[38;5;241m=[39m[38;5;124mf[39m[38;5;124m'[39m[38;5;132;01m{[39;00mperiod[38;5;132;01m}[39;00m[38;5;124m yr RL[39m[38;5;124m'[39m, color[38;5;241m=[39m[43mcolors[49m[i],linewidth[38;5;241m=[39m[38;5;241m0.5[39m)
[1;32m     19[0m [38;5;66;03m# add bold black line at 0 for MHHW[39;00m
[1;32m     20[0m ax[38;5;241m.[39maxhline([38;5;241m0[39m, color[38;5;241m=[39m[38;5;124m'[39m[38;5;124mblack[39m[38;5;124m'[39m, linewidth[38;5;241m=[39m[38;5;241m2[39m)

[0;31mNameError[0m: name 'colors' is not defined

